client/dashboard/app/queries/site-staging-sites.ts

```diff
@@ -1,9 +1,32 @@
-import { createStagingSite, deleteStagingSite } from '../../data/site-staging-site';
+import {
+	createStagingSite,
+	deleteStagingSite,
+	fetchStagingSiteOf,
+	StagingSite,
+} from '../../data/site-staging-site';
+import { queryClient } from '../query-client';
+export const STAGING_SITE_DELETE_MUTATION_KEY = 'staging-site-delete-mutation';
+
+export const hasStagingSiteQuery = ( productionSiteId: number ) => ( {
+	queryKey: [ 'staging-site', productionSiteId ],
```

Comment:
It's misleading as we're using `staging-site` prefix in `isDeletingStagingSiteQuery` with a staging site ID but here we're using a production one. Suggestion:


```suggestion
	queryKey: [ 'site', productionSiteId, 'has-staging-site' ],
```